### PR TITLE
Enhancement master reduce number of layers

### PR DESCRIPTION
**Pull request by sbts**
- Reduce the number of AUFS layers
- improve handling of apt commands to prevent caching issues, and possibly reduce the container size

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -25,21 +22,23 @@ RUN DEBIAN_FRONTEND=noninteractive && \
...
 ENV NODE_PATH /usr/local/lib/node_modules
-ENV DEBIAN_FRONTEND=noninteractive
 
 ARG CACHEBUST
 
 
 # Java & Nodejs for doing Dojo build
 # Uglify needs to be installed right before 'make dojo'?!
-RUN apt-get -y install git make gcc libperl-dev npm curl && \
+RUN apt-get update && \
```
Comment (freelock) [inline Dockerfile]: What about moving all of this run command other than the clone to the previous run line, and eliminate one apt-get update? If we're putting a cache-bust here, we might end up needing a dist-upgrade...

File: Dockerfile [hunk 2]
```diff
@@ -55,7 +54,8 @@ RUN apt-get -y install git make gcc libperl-dev npm curl && \
     rm -rf /usr/local/lib/node_modules && \
     apt-get autoremove -y && \
     apt-get autoclean && \
-    rm -rf ~/.cpanm
+    rm -rf ~/.cpanm && \
+    rm -rf /var/lib/apt/lists/*
```
Comment (freelock) [inline Dockerfile]: This looks redundant... would be curious whether the result is significant of removing this at all, but seems like it has already been done twice...

File: Dockerfile [hunk 3]
```diff
@@ -1,12 +1,9 @@
...
 MAINTAINER  Freelock john@freelock.com
 
-RUN echo "APT::Install-Recommends \"false\";\nAPT::Install-Suggests \"false\";\n" > /etc/apt/apt.conf.d/00recommends
-
-
 # Install Perl, Tex, Starman, psql client, and all dependencies
-RUN DEBIAN_FRONTEND=noninteractive && \
-  apt-get update && apt-get -y install \
+RUN echo -e "APT::Install-Recommends \"false\";\nAPT::Install-Suggests \"false\";\n" > /etc/apt/apt.conf.d/00recommends && \
+  apt-get update DEBIAN_FRONTEND="noninteractive" apt-get -y install && \
```
Comment (ehuelsmann) [inline Dockerfile]: There's a syntax error here: no command separator.

File: Dockerfile [hunk 4]
```diff
@@ -25,20 +22,30 @@ RUN DEBIAN_FRONTEND=noninteractive && \
...
 # Uglify needs to be installed right before 'make dojo'?!
-RUN apt-get -y install git make gcc libperl-dev npm curl && \
+
+# These packages are only needed during the dojo build
+ENV DOJO_Build_Deps git make gcc libperl-dev npm curl
+# These packages can be removed after the dojo build
+ENV DOJO_Build_Deps_removal ${DOJO_Build_Deps} nodejs
+
+RUN apt-get update && \
+    DEBIAN_FRONTEND="noninteractive" apt-get -y install ${DOJO_Build_Deps} && \
```
Comment (ehuelsmann) [inline Dockerfile]: this debian frontend setting applies only to a single command, but all `apt-get` invocations need it.

**Pull request merged**